docs/v1/P4Runtime-Spec.mdk

```diff
@@ -1456,41 +1456,132 @@ types (32-bit and 64-bit words). The P4 language does not put any limit on the
+binary string bytes required to represent the maximum integer of `p4_bitwidth`
+bits as:
+~ Begin CPP
+binary_string_bytes = (p4_bitwidth + 7) / 8;
+~ End CPP
+The server's decision
+to accept or reject the binary string depends on the string's content and its
+actual length in the P4Runtime request, as follows:
+
+* If the string's actual byte length is shorter than `binary_string_bytes`
+  but greater than zero, the server always accepts the string and treats the
+  missing bytes as zeroes in the most-significant positions.
+
+* If the string's actual byte length is longer than `binary_string_bytes`, the
+  server rejects any string where the extra high-order bytes contain non-zero
+  values.  Otherwise, the server truncates the extra bytes from the front
+  of the string and processes the remainder of the string as if it contains
+  exactly `binary_string_bytes`, as described below.
+
+* If the string's original or truncated byte length is equal to
+  `binary_string_bytes`, then the server must also consider the P4 integer
+  type's byte alignment.  Byte-aligned strings that satisfy the expression:
+~ Begin CPP
+  (binary_string_bytes * 8 == p4_bitwidth)
+~ End CPP
+  are always valid.  For non-aligned strings, the server requires
+  the client to fill the high-order pad bits with zeroes, and the server
+  rejects any binary strings with non-zero pad bits.
+
+* If the string's byte length is zero, the server always rejects the string.
+
+When the server rejects a binary string due to any of the previous criteria,
+it returns an `OUT_OF_RANGE` error.
+
+For all binary strings, P4Runtime uses big-endian (&ie; network) byte-order.
 For signed integer values (`int<W>` P4 type), P4Runtime uses the same two's
-complement bitwise representation as P4.
-
-Table [#tab-exmpl-bytestring-encoding] shows examples of integer values encoded
-as valid binary strings for P4Runtime.
-
-~ TableFigure { #tab-exmpl-bytestring-encoding; \
-    caption: "Examples of Bytestring Encoding"; }
-|-----------|----------------|-------------------------|
-| P4 type   | Integer value  | P4Runtime binary string |
-+-----------+----------------+-------------------------+
-| `bit<8>`  | 99 (0x63)      | `\x63`                  |
-| `bit<16>` | 99 (0x63)      | `\x00\x63`              |
-| `bit<16>` | 12388 (0x3064) | `\x30\x64`              |
-| `bit<12>` | 99 (0x63)      | `\x00\x63`              |
-| `int<8>`  | 99 (0x63)      | `\x63`                  |
-| `int<8>`  | -99 (-0x63)    | `\x9d`                  |
-| `int<12>` | -739 (-0x2e3)  | `\x0d\x1d`              |
-| `int<16>` | 0 (0x0)        | `x00\x00`               |
-|-----------|----------------|-------------------------|
+complement bitwise representation as P4.  Table [#tab-valid-bytestring-encoding]
+shows various examples of integer values that the server accepts as valid
+P4Runtime binary strings according to the criteria in the list above.
+
+~ TableFigure { #tab-valid-bytestring-encoding; \
```

Comment:
I checked that table and it renders fine in PDF, although we are almost at the limit :)

Reply:
Yes - I rendered in PDF before creating the pull request.